packages/mint-components/src/components/sqm-referral-codes/sqm-referral-codes.tsx

```diff
@@ -0,0 +1,110 @@
+import { isDemo } from "@saasquatch/component-boilerplate";
+import { withHooks } from "@saasquatch/stencil-hooks";
+import { Component, Prop, h } from "@stencil/core";
+import deepmerge from "deepmerge";
+import { DemoData } from "../../global/demo";
+import { getProps } from "../../utils/utils";
+import {
+  ReferralCodesView,
+  ReferralCodesViewProps,
+} from "./sqm-referral-codes-view";
+import { useReferralCodes } from "./useReferralCodes";
+
+/**
+ * @uiName Referral Codes
+ * @slots [{"name":"", "title":"Promo Codes Content"}]
```

Comment:
@slots seem incomplete

Reply:
In this case it looks like the loading and empty state are hard-coded and imported, might be a question of whether we should support both